### PR TITLE
Ganesh/docissue

### PR DESCRIPTION
**Pull request by Ganeshshinde-2003**
The above command is a one liner command that will easily install all the Python build dependencies for your Ubuntu or Debian operating system. It installs all the packages that are mentioned in the previous comment and it will save time instead of running each command separately. 
The command as provided would not work for some systems some time (even for me is wasn't working) because of the extra spaces. To run the command correctly, you will need to remove all the spaces between the package names and keep only one space after "install" keyword. 

@seanlip @U8NWXD

### CODE REVIEW

**Comment by U8NWXD** [comment]
> To run the command correctly, you will need to remove all the spaces between the package names and keep only one space after "install" keyword

Do you mean that instead of running `sudo apt-get install make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget llvm libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev` (the command you have in this PR) sometimes you have to run `sudo apt-get install makebuild-essentiallibssl-dev zlib1g-devlibbz2-devlibreadline-devlibsqlite3-devwgetllvmlibncursesw5-devxz-utilstk-devlibxml2-devlibxmlsec1-devlibffi-devliblzma-dev` (the same command but with spaces between the package names removed)? That doesn't seem right

**Comment by Ganeshshinde-2003** [comment]
NO sir, we can run the command. (```sudo apt-get install make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget llvm libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev```) It will also give you the same results you wanted.
If you want it the way you suggested in your previous comment (```sudo apt-get install makebuild-essentiallibssl-dev zlib1g-devlibbz2-devlibreadline-devlibsqlite3-devwgetllvmlibncursesw5-devxz-utilstk-devlibxml2-devlibxmlsec1-devlibffi-devliblzma-dev```). 
Can I be the one who changes it. 
please
Thank you sir

**Comment by U8NWXD** [comment]
OK I'm still confused by this:

> To run the command correctly, you will need to remove all the spaces between the package names and keep only one space after "install" keyword.

Can you clarify what you meant?

**Comment by Ganeshshinde-2003** [comment]
Using these commands some times won't end up installing the packages and it will stop installing . It happened to me more than 5 times sir :
```sudo apt-get install make
sudo apt-get install build-essential
sudo apt-get install libssl-dev
sudo apt-get install zlib1g-dev
sudo apt-get install libbz2-dev
sudo apt-get install libreadline-dev
sudo apt-get install libsqlite3-dev
sudo apt-get install wget
sudo apt-get install llvm
sudo apt-get install libncursesw5-dev
sudo apt-get install xz-utils
sudo apt-get install tk-dev
sudo apt-get install libxml2-dev
sudo apt-get install libxmlsec1-dev
sudo apt-get install libffi-dev
sudo apt-get install liblzma-dev

**Comment by Ganeshshinde-2003** [comment]
The main point is for some reasons it didn't work for me and I tried many times. That's why I used this command and It is shorter one :
```sudo apt-get install make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget llvm libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev``` 
or
 ```sudo apt-get install makebuild-essentiallibssl-dev zlib1g-devlibbz2-devlibreadline-devlibsqlite3-devwgetllvmlibncursesw5-devxz-utilstk-devlibxml2-devlibxmlsec1-devlibffi-devliblzma-dev```

**Comment by Ganeshshinde-2003** [comment]
Is my pull request accepted or do I need to change it according to you @U8NWXD 
@seanlip @gp201

**Comment by gp201** [comment]
@Ganeshshinde-2003 can you provide the O/P for running both commands?

I believe both commands do the same task and shouldn't fail in one case and succeed in the other.

**Comment by U8NWXD** [comment]
> or
> `sudo apt-get install makebuild-essentiallibssl-dev zlib1g-devlibbz2-devlibreadline-devlibsqlite3-devwgetllvmlibncursesw5-devxz-utilstk-devlibxml2-devlibxmlsec1-devlibffi-devliblzma-dev`

I don't think this command will work. `apt-get install` operates on space-separated package names, and we want to install `make`, not `makebuild-essentiallibssl-dev`.

@Ganeshshinde-2003 when the multi-line approach failed, were you pasting all the lines at once, or were you running each line one at a time?

**Comment by U8NWXD** [comment]
We're not ready to accept your PR yet. We need to understand what problem it's solving first, and I'm still confused

**Comment by Ganeshshinde-2003** [comment]
Hey @U8NWXD ,
I pasted all the lines at once. But it wasn't installing the packages. and I tried many times by pasting all at once but it wasn't helping. and Here is the proof 
image
And Running the commands line by line is time consuming process sir.

**Comment by Ganeshshinde-2003** [comment]
@U8NWXD 
@gp201 

Screenshot (115)

**Comment by U8NWXD** [comment]
> I pasted all the lines at once.

Ahh this is the problem. Linux command lines (and the command lines of most OSes) expect each command to be entered one at a time, so you'll need to paste the commands one line at a time.

Instead of changing the installation commands, how about we add a note on the [installation landing page](https://github.com/oppia/oppia/wiki/Installing-Oppia) explaining that commands need to be entered one at a time? It can go along with the other notation clarifications we have there

**Comment by Ganeshshinde-2003** [comment]
Yaa we can do that  @U8NWXD 
But sir, Instead of running all the commands one by one. We can use one line command it will reduce the time

**Comment by U8NWXD** [comment]
Sure. Let me know when you've updated this PR with the changes we've discussed

**Comment by Ganeshshinde-2003** [comment]
@U8NWXD 
I have updated this PR. Please see to it if I need to make any changes

File: Installing-Oppia-(Linux;-Python-3).md [hunk 1]
```diff
@@ -94,24 +94,9 @@ bash scripts/install_prerequisites.sh
...
-    sudo apt-get install xz-utils
-    sudo apt-get install tk-dev
-    sudo apt-get install libxml2-dev
-    sudo apt-get install libxmlsec1-dev
-    sudo apt-get install libffi-dev
-    sudo apt-get install liblzma-dev
-    ```
+
+
+    ```sudo apt-get install make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget llvm libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev```
```
Comment (U8NWXD) [inline Installing-Oppia-(Linux;-Python-3).md]: Since this is a single-line code block, you should use `` ` `` instead of ` ``` `

**Review by U8NWXD — CHANGES_REQUESTED** [review]

File: Installing-Oppia-(Linux;-Python-3).md (hunk 1 above)
Comment (Ganeshshinde-2003) [reply Installing-Oppia-(Linux;-Python-3).md]: @U8NWXD  I changed it from ``` to `

**Comment by Ganeshshinde-2003** [comment]
@U8NWXD
I made changes in PR that you asked for
From ``` to `

**Review by U8NWXD — COMMENTED** [review]
Thanks @Ganeshshinde-2003!

**Review by U8NWXD — APPROVED** [review]

**Comment by U8NWXD** [comment]
@seanlip can you merge this after confirming that Ganeshshinde-2003 has signed the CLA?

**Comment by seanlip** [comment]
Yup, looks good. Merging, congrats @Ganeshshinde-2003 on your first contribution!

**Comment by seanlip** [comment]
Also @Ganeshshinde-2003, next time, please use a more descriptive title for your PR. See instructions [here](https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-4-create-a-pull-request) (third bullet point). 

Thanks!

**Comment by Ganeshshinde-2003** [comment]
Thank you! @U8NWXD  @seanlip @gp201 
For recognition of my first contribution

**Pull request merged**